pandas/io/formats/excel.py

```diff
@@ -466,14 +467,13 @@ def _format_header_mi(self):
             coloffset = len(self.df.index[0]) - 1
 
         if self.merge_cells:
-            # Format multi-index as a merged cells.
-            for lnum in range(len(level_lengths)):
-                name = columns.names[lnum]
-                yield ExcelCell(lnum, coloffset, name, self.header_style)
-
             for lnum, (spans, levels, level_codes) in enumerate(
                 zip(level_lengths, columns.levels, columns.codes)
             ):
+                # Format multi-index as a merged cells.
```

Comment:
Are these changes related to the constant memory issue?

Reply:
yes, the constant memory issue is about the order of generating ExcelCell. So sometimes simply reorder the yield sequence can solve the problem.